lib/mongoid/criteria/queryable/smash.rb

```diff
@@ -74,8 +74,12 @@ def [](key)
         # @return [ String ] The normalized key.
         #
         # @since 1.0.0
-        def normalized_key(name, serializer)
-          serializer && serializer.localized? ? "#{name}.#{::I18n.locale}" : name
+        def normalized_key(name, serializer, opts = {})
+          if serializer && serializer.localized? && opts[:localize] != false
```

Comment:
Reading my earlier comments. Rather than `localize: true` we should call it `i18n_fallbacks: false`